parts/k8s/cloud-init/artifacts/cse_main.sh

```diff
@@ -229,16 +229,25 @@ fi
 VALIDATION_ERR=0
 
 {{- if IsHostedMaster }}
-RES=$(retrycmd_if_failure 20 1 3 nslookup ${API_SERVER_NAME})
+API_SERVER_DNS_RETRIES=20
+if [[ $API_SERVER_NAME == *.privatelink.* ]]; then
+  API_SERVER_DNS_RETRIES=100
```

Comment:
is this enough? I read the email discussion that the dns resolution failed for 10minutes, is that not the cae?

Reply:
Good catch, 200 should be here and 100 in the later check.